SSPSolution/SSPSolution/Player.cpp

```diff
@@ -93,9 +93,9 @@ int Player::Update(float dT, InputHandler* inputHandler)
 			
 			if (!stateExists(RAGDOLL_STATE))
 			{
-				this->m_oldAnimState = this->m_aComp->previousState;
+				this->m_aComp->previousState = this->m_aComp->currentState;
 				SetAnimationComponent(RAGDOLL_STATE, 0.f, Blending::NO_TRANSITION, false, false, 0.f, 1.0);	
-				this->m_aComp->previousState = RAGDOLL_STATE;
+				this->m_aComp->currentState = RAGDOLL_STATE;
```

Comment:
Since the ragdoll states is also enumeration and modified in this pull request I would recommend that you add the enumeration name for these too.